From bd05332bf5ec4ce24f5e5a831fad783e39d8a581 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Sun, 1 Sep 2013 13:54:41 -0500
Subject: [PATCH 11/13] crypto/bio: Disable xopenlog()/xcloselog() on Phar Lap
 ETS

Disable xopenlog()/xcloselog() on Phar Lap ETS because NI's distribution
does not support the requisite system APIs.

Upstream-Status: Inappropriate [Phar Lap ETS support]
---
 crypto/bio/bss_log.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/crypto/bio/bss_log.c b/crypto/bio/bss_log.c
index 1283a525b4..3d9168f30c 100644
--- a/crypto/bio/bss_log.c
+++ b/crypto/bio/bss_log.c
@@ -283,9 +283,11 @@ static int MS_CALLBACK slg_puts(BIO *bp, const char *str)
 
 static void xopenlog(BIO *bp, char *name, int level)
 {
+#ifndef PHARLAP
     if (check_winnt())
         bp->ptr = RegisterEventSourceA(NULL, name);
     else
+#endif /* PHARLAP */
         bp->ptr = NULL;
 }
 
@@ -331,8 +333,10 @@ static void xsyslog(BIO *bp, int priority, const char *string)
 
 static void xcloselog(BIO *bp)
 {
+#ifndef PHARLAP
     if (bp->ptr)
         DeregisterEventSource((HANDLE) (bp->ptr));
+#endif /* PHARLAP */
     bp->ptr = NULL;
 }
 
-- 
2.13.1

